### PR TITLE
Register Liquibase Precondition services

### LINKED ISSUE #13839: Liquibase failing dbms-type pre-condition in native build
**Describe the bug**

I have a liquibase script with a precondition on the dbms type:

```
<preConditions>
   <dbms type="postgresql,h2"/>
</preConditions>
```

When runing on a JVM it works correctly. When running as native build, following error is thrown:

```
Dec 11, 2020 1:29:31 PM liquibase.changelog
INFO: Reading from qumearecording.databasechangelog
Dec 11, 2020 1:29:31 PM io.quarkus.runtime.ApplicationLifecycleManager run
ERROR: Failed to start application (with profile prod)
liquibase.exception.ValidationFailedException: Validation Failed:
     1 preconditions failed
     db/changeLog.xml : DBMS Precondition failed: expected null, got postgresql

        at liquibase.changelog.DatabaseChangeLog.validate(DatabaseChangeLog.java:298)
        at liquibase.changelog.DatabaseChangeLog.validate(DatabaseChangeLog.java:274)
        at liquibase.Liquibase.validate(Liquibase.java:2154)
        at io.quarkus.liquibase.runtime.LiquibaseRecorder.doStartActions(LiquibaseRecorder.java:48)
        at io.quarkus.deployment.steps.LiquibaseProcessor$createBeansAndStartActions-960013254.deploy_0(LiquibaseProcessor$createBeansAndStartActions-960013254.zig:80)
        at io.quarkus.deployment.steps.LiquibaseProcessor$createBeansAndStartActions-960013254.deploy(LiquibaseProcessor$createBeansAndStartActions-960013254.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:455)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:96)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:62)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:104)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
```

I also tested that the dbms attribute on the changeSet is working correctly. 

**To Reproduce**

Run a project with contains a liquibase <preCondition> native.

**Environment:**
 - Quarkus version: 1.10.3.Final

### PR DESCRIPTION
**Pull request by machi1990**
Fixes https://github.com/quarkusio/quarkus/issues/13839

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Thanks!

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java [hunk 1]
```diff
@@ -169,7 +170,7 @@ private void addService(BuildProducer<ServiceProviderBuildItem> services,
                     service);
             services.produce(new ServiceProviderBuildItem(serviceClass.getName(), implementations.toArray(new String[0])));
 
-            reflective.produce(new ReflectiveClassBuildItem(true, true, false, implementations.toArray(new String[0])));
+            reflective.produce(new ReflectiveClassBuildItem(true, true, true, implementations.toArray(new String[0])));
```
Comment (geoand) [inline extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java:173]: Maybe it makes sense to only have `Precondition` do this and keep the others like they are?

Comment (machi1990) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java:173]: We can do that. I thought so too but I think the reason I went with this implementation is that the others do not have fields that are set reflectively.

Comment (machi1990) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java:173]: We can make it explicit and it won't be a harm.

Comment (machi1990) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java]: Done

**Review by geoand — APPROVED** [review]

**Pull request merged**